include/aws/crt/iot/DeviceDefender.h

```diff
@@ -0,0 +1,184 @@
+#pragma once
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/crt/io/EventLoopGroup.h>
+#include <aws/crt/mqtt/MqttClient.h>
+
+#include <aws/iotdevice/device_defender.h>
+
+namespace Aws
+{
+    namespace Crt
+    {
+
+        namespace Io
+        {
+            class EventLoopGroup;
+        }
+
+        namespace Mqtt
+        {
+            class MqttConnection;
+        }
+
+        namespace Iot
+        {
+
+            /**
+             * Invoked upon DeviceDefender V1 task cancellation.
+             */
+            using OnDefenderV1TaskCancelledHandler = std::function<void(void *)>;
+
+            using DeviceDefenderReportFormat = aws_iotdevice_defender_report_format;
+
+            /**
+             * Enum used to expose the status of a DeviceDefenderV1 task.
+             */
+            enum class DeviceDefenderV1ReportTaskStatus
+            {
+                Ready = 0,
+                Running = 1,
+                Stopped = 2,
+                Failed = 3,
+            };
+
+            /**
+             * Represents a unique configuration for a device defender V1 report task. You can use a single instance of
+             * this class PER task you want to create.
+             */
+            class AWS_CRT_CPP_API DeviceDefenderV1ReportTaskConfig final
+            {
+
+                friend class DeviceDefenderV1ReportTask;
+                friend class DeviceDefenderV1ReportTaskConfigBuilder;
+
+              public:
+                /**
+                 * Creates a client configuration for use with making new AWS Iot specific MQTT Connections with MTLS.
+                 */
+                DeviceDefenderV1ReportTaskConfig(
+                    std::shared_ptr<Mqtt::MqttConnection> mqttConnection,
+                    ByteCursor thingName,
+                    Io::EventLoopGroup &eventLoopGroup,
+                    DeviceDefenderReportFormat reportFormat,
+                    uint64_t taskPeriodNs,
+                    uint64_t networkConnectionSamplePeriodNs,
+                    OnDefenderV1TaskCancelledHandler &&onCancelled = NULL,
+                    void *cancellationUserdata = nullptr) noexcept;
+
+              private:
+                OnDefenderV1TaskCancelledHandler onCancelled;
+                void *cancellationUserdata;
+                aws_iotdevice_defender_report_task_config taskConfig;
+            };
+
+            /**
+             * Represents a builder for creating a DeviceDefenderV1ReportTaskConfig object.
+             */
+            class AWS_CRT_CPP_API DeviceDefenderV1ReportTaskConfigBuilder final
```

Comment:
I am wondering if this makes sense:

1. Remove DeviceDefenderV1ReportTaskConfig and DeviceDefenderV1ReportTaskConfigBuilder.
1. Create DeviceDefenderV1ReportTaskBuilder
1. The TaskBuilder interface would look similar with ConfigBuilder.
1. The usage would be something like:
```
TaskBuilder taskBuilder;
Task t = taskBuilder.withA().withB().Build();
```

So the idea is to remove the exposure of config altogether.

Reply:
BTW, I have **NOT** checked what is the existing pattern for the rest of the CRT.

Reply:
Yeah I think that could make sense. I've thought about that pattern as well and hiding the config within the Task class so that Task is the only exposed class. The builder pattern could still ensure that the Task is given all required parameters, and that might clean things up. I'll give that a try with a follow up commit. Thanks!